enclave/chain_relay/src/lib.rs

```diff
@@ -247,44 +247,51 @@ impl Validator for LightValidation {
 		ancestry_proof: Vec<Header>,
 		validator_set: AuthorityList,
 		validator_set_id: SetId,
-		grandpa_proofs: Option<Justifications>,
+		justifications: Option<Justifications>,
 	) -> Result<(), Error> {
 		let mut relay = self.tracked_relays.get_mut(&relay_id).ok_or(Error::NoSuchRelayExists)?;
 
 		// Check that the new header is a decendent of the old header
 		let last_header = &relay.last_finalized_block_header;
 		Self::verify_ancestry(ancestry_proof, last_header.hash(), &header)?;
 
-		if grandpa_proofs.is_none() {
-			relay.last_finalized_block_header = header.clone();
-			relay.unjustified_headers.push(header.hash());
-			debug!(
-				"Syncing finalized block without grandpa proof. Amount of unjustified headers: {}",
-				relay.unjustified_headers.len()
-			);
-			return Ok(())
-		}
-
-		let block_hash = header.hash();
-		let block_num = *header.number();
-
 		// Check that the header has been finalized
 		let voter_set =
 			VoterSet::new(validator_set.clone().into_iter()).expect("VoterSet may not be empty");
 
-		// https://github.com/paritytech/substrate/pull/7640/files
-		// multiple proofs due to multiple consensus protococol possible..
-		for grandpa_proof in grandpa_proofs.unwrap().iter() {
-			Self::verify_grandpa_proof::<Block>(
-				// https://github.com/paritytech/substrate/pull/7640/files
-				// only works for one consenus protocol. In case of multiple
-				// use .into_justification(consensusprotocol) of substrate
-				grandpa_proof.clone(),
-				block_hash,
-				block_num,
-				validator_set_id,
-				&voter_set,
-			)?;
+		// ensure justifications is a grandpa justification
+		let grandpa_justification =
+			justifications.and_then(|just| just.into_justification(GRANDPA_ENGINE_ID));
```

Comment:
What does `into_justification` do?

Reply:
https://github.com/paritytech/substrate/blob/master/primitives/runtime/src/lib.rs#L141-L145

Reply:
I have added it to match substrates import implementation: https://github.com/paritytech/substrate/blob/master/client/finality-grandpa/src/import.rs#L628-L629

Reply:
Ah, ok i see. It will make the justification into a None, if it is something different.